@here/harp-materials/lib/SolidLineMaterial.ts

```diff
@@ -652,21 +623,17 @@ export class SolidLineMaterial extends RawShaderMaterial
     }
 
     /**
-     * Line opacity.
+     * To set the opacity inclusive updating the uniforms and stencilWrite.
      */
-    get opacity(): number {
-        return this.m_opacity;
-    }
-
-    set opacity(value: number) {
-        this.m_opacity = value;
+    setOpacity(value: number) {
+        this.opacity = value;
         // Setting opacity before uniform being created requires late invalidation,
         // call to invalidateOpacity() is done at the end of c-tor.
         if (this.uniforms?.opacity) {
             this.uniforms.opacity.value = value;
         }
 
-        this.stencilWrite = this.m_opacity < 0.98;
+        this.stencilWrite = value < 0.98;
```

Comment:
As I mentioned, setOpacity could be moved to RawShaderMaterial. If this stencilWrite flag is specific of SoliLineMaterial,override setOpacity here.

Reply:
done